tools/MultiProjectValidator/ProjectLoader.cs

```diff
@@ -29,10 +30,23 @@ private static List<ProjectContext> LoadProjectContexts(string[] projectFiles)
             {
                 var fileTargetContexts = ProjectContext.CreateContextForEachTarget(file);
 
-                projectContexts.AddRange(fileTargetContexts);    
+                foreach(var context in fileTargetContexts)
+                {
+                    if (!TargetIsDesktop(context))
+                    {
+                        projectContexts.Add(context);
+                    }
+                }  
             }
 
             return projectContexts;
         }
+
+        private static bool TargetIsDesktop(ProjectContext context)
```

Comment:
Yeah I see what you were saying here. 

This portion of the code is responsible for loading all of the contexts and shouldn't be making a decision on which ones should/shouldn't be validated.

I don't know if an extension method is the right place either. To me it would make more sense if the individual Analysis Rules determined whether or not they should be skipping a context. So I would add this logic to https://github.com/dotnet/cli/blob/master/tools/MultiProjectValidator/AnalysisRules/DependencyMismatchRule.cs

Thoughts?